tests/quantization/finegrained_fp8/test_finegrained_fp8.py

```diff
@@ -0,0 +1,627 @@
+        weight_scale_inv = quantized_model.transformer_blocks[0].ff.net[2].weight_scale_inv
+
+        self.assertTrue(isinstance(layer, FP8Linear))
+
+        self.assertEqual(weight_scale_inv.dtype, torch.bfloat16)
+
+        self.assertEqual(weight.dtype, torch.float8_e4m3fn)
+
+    def test_device_map_auto(self):
+        """
+        Test if the quantized model is working properly with "auto"
+        """
+
+        inputs = self.get_dummy_tensor_inputs(torch_device)
+        # requires with different expected slices since models are different due to offload (we don't quantize modules offloaded to cpu/disk)
+        expected_slice_auto = np.array(
+            [ 
+                0.34375,
+                -0.0402832,
+                0.01226807,
+                -0.22851562,
+                -0.49414062,
+                0.4375,
+                -0.16992188,
+                -0.66015625,
+                0.43164062
+            ]
+        )
+
+        quantization_config = FinegrainedFP8Config(
+            modules_to_not_convert=["x_embedder", "proj_out"], 
+            weight_block_size=(32, 32)
+        )
+        quantized_model = FluxTransformer2DModel.from_pretrained(
+            "hf-internal-testing/tiny-flux-pipe",
+            subfolder="transformer",
+            quantization_config=quantization_config,
+            device_map="auto",
+            torch_dtype=torch.bfloat16,
+        )
+
+        output = quantized_model(**inputs)[0]
+        output_slice = output.flatten()[-9:].detach().float().cpu().numpy()
+
+        self.assertTrue(numpy_cosine_similarity_distance(output_slice, expected_slice_auto) < 1e-3)
+
+
+    def test_modules_to_not_convert(self):
+        quantization_config = FinegrainedFP8Config(weight_block_size=(32, 32), modules_to_not_convert=["transformer_blocks.0", "proj_out", "x_embedder"])
+        quantized_model_with_not_convert = FluxTransformer2DModel.from_pretrained(
+            "hf-internal-testing/tiny-flux-pipe",
+            subfolder="transformer",
+            quantization_config=quantization_config,
+            torch_dtype=torch.bfloat16,
+            device_map=torch_device,
+        )
+
+        unquantized_layer = quantized_model_with_not_convert.transformer_blocks[0].ff.net[2]
+        self.assertTrue(isinstance(unquantized_layer, torch.nn.Linear))
+        self.assertEqual(unquantized_layer.weight.dtype, torch.bfloat16)
+
+        quantization_config = FinegrainedFP8Config(
+            modules_to_not_convert=["x_embedder", "proj_out"], 
+            weight_block_size=(32, 32)
+        )
+        quantized_model = FluxTransformer2DModel.from_pretrained(
+            "hf-internal-testing/tiny-flux-pipe",
+            subfolder="transformer",
+            quantization_config=quantization_config,
+            torch_dtype=torch.bfloat16,
+            device_map=torch_device,
+        )
+
+        size_quantized_with_not_convert = self.get_model_size_in_bytes(quantized_model_with_not_convert)
```

Comment:
Can't we leverage ?
https://github.com/huggingface/diffusers/blob/f0dba33d82af991369806312f61ab4c6cb7a8dd1/src/diffusers/models/modeling_utils.py#L1721